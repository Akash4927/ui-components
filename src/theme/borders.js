import { kebabCase, forEach } from 'lodash';

export const borderRadius = {
  none: '0',
  soft: '2px',
  circle: '50%',
};

// Collects all theme border radii as SCSS vars.
export function themeBorderRadiiAsScss() {
  const themeBorderRadii = [];

  forEach(borderRadius, (value, name) => {
    themeBorderRadii.push(`$border-radius-${kebabCase(name)}: ${value}`);
  });

  return themeBorderRadii;
}
