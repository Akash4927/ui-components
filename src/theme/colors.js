import { isString, kebabCase, forEach } from 'lodash';

export const colors = {
  // uncategorized colors
  aliceBlue: '#f0f8ff',
  amethystSmoke: '#9292b7',
  athens: '#e2e2ec',
  athensGray: '#eeeef4',
  black: '#1a1a1a',
  blueHaze: '#c0c0d5',
  comet: '#5b5b88',
  cometDark: '#5a5a7b',
  doveGray: '#737373',
  eastBay: '#4a5d87',
  gallery: '#f4f4f4',
  gunpowder: '#3c3c5b',
  kimberly: '#7d7da8',
  kimberlyDark: '#656597',
  logan: '#b1b1cb',
  mischka: '#dfdfea',
  mulledWine: '#46466a',
  mystic: '#dae4ea',
  silver: '#ccc',
  turquoise: '#037aa9',
  white: '#fff',
  whiteLilac: '#f0f0fa',
  whiteSmoke: '#fafafc',
  promQL: {
    /**
     * GHColors theme by Avi Aryan (http://aviaryan.in)
     * Inspired by Github syntax coloring
     */
    comment: '#bbbbbb',
    string: '#e3116c',
    punctuation: '#393a34',
    entity: '#36acaa',
    metricName: '#2aa198',
    attrName: '#00a4db',
    function: '#dc322f',
    deleted: '#9a050f',
    tag: '#00009f',
    // Dropdown colors,
    salmon: '#ff7c7c',
  },
  // use these for new stuff!
  primary: {
    charcoal: '#32324b',
    lavender: '#8383ac',
  },
  accent: {
    blue: '#00d2ff',
    orange: '#ff4b19',
  },
  status: {
    // Use these when colorizing elements that should indicate a state or outcome, ie Alerts.
    success: '#38bd93', // green
    error: '#c7254e', // red
    warning: '#d4ab27', // amber
    info: '#049cd7', // blue
    disabled: '#9a9a9a',
  },
  // Used by PrometheusGraph component
  graphThemes: {
    blue: [
      '#c7e9b4',
      '#7ecdbb',
      '#1eb5eb',
      '#1d91bf',
      '#235fa9',
      '#253393',
      '#084181',
    ],
    purple: [
      '#c1d4e7',
      '#9fbddb',
      '#8d95c6',
      '#8282ab',
      '#89429e',
      '#800f7a',
      '#0b0533',
    ],
    mixed: [
      '#c7e9b4',
      '#c1d4e7',
      '#7ecdbb',
      '#9fbddb',
      '#1eb5eb',
      '#8d95c6',
      '#1d91bf',
      '#8282ab',
      '#235fa9',
      '#89429e',
      '#253393',
      '#800f7a',
      '#084181',
      '#0b0533',
    ],
  },
  // Third-party specific colors - not to be used in the theme!
  thirdParty: {
    // Google single-click login
    cornflowerBlue: '#4285f4',
    azure: '#3769bb',
  },
};

// Flattens and collects all theme colors as SCSS vars
export function themeColorsAsScss() {
  const ignoreKeys = ['graphThemes'];
  const themeColors = [];

  forEach(colors, (value, name) => {
    const colorPrefix = `$color-${kebabCase(name)}`;
    if (ignoreKeys.includes(name)) return;

    if (isString(value)) {
      themeColors.push(`${colorPrefix}: ${value}`);
    } else {
      forEach(value, (innerValue, subname) => {
        if (ignoreKeys.includes(subname)) return;
        themeColors.push(`${colorPrefix}-${kebabCase(subname)}: ${innerValue}`);
      });
    }
  });

  return themeColors;
}
