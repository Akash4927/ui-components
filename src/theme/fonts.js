import { kebabCase, forEach } from 'lodash';

export const fontFamilies = {
  regular: "'proxima-nova', Helvetica, Arial, sans-serif",
  monospace: "'Roboto Mono', monospace",
};

export const fontSizes = {
  huge: '48px',
  extraLarge: '32px',
  large: '22px',
  normal: '16px',
  small: '14px',
  tiny: '12px',
};

// Collects all theme font vars as SCSS vars.
export function themeFontsAsScss() {
  const themeFonts = [];

  forEach(fontFamilies, (value, name) => {
    themeFonts.push(`$font-family-${kebabCase(name)}: ${value}`);
  });

  forEach(fontSizes, (value, name) => {
    themeFonts.push(`$font-size-${kebabCase(name)}: ${value}`);
  });

  return themeFonts;
}
