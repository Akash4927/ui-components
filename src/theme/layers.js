import { kebabCase, forEach } from 'lodash';

export const layers = {
  front: 1,
  toolbar: 2,
  alert: 3,
  notification: 4,
  dropdown: 5,
  tooltip: 6,
  modal: 7,
};

// Collects all theme z-index layers as SCSS vars
export function themeLayersAsScss() {
  const themeLayers = [];

  forEach(layers, (value, name) => {
    themeLayers.push(`$layer-${kebabCase(name)}: ${value}`);
  });

  return themeLayers;
}
