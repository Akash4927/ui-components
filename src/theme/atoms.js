import { darken } from 'polished';

import { colors } from './colors';

// component-specific
export const atoms = {
  Alert: {
    default: {
      color: colors.black,
      background: colors.gallery,
      borderColor: darken(0.15, colors.gallery),
    },
    info: {
      color: colors.white,
      background: colors.status.info,
      borderColor: darken(0.15, colors.status.info),
    },
    success: {
      color: colors.white,
      background: colors.status.success,
      borderColor: darken(0.15, colors.status.success),
    },
    warning: {
      color: colors.white,
      background: colors.status.warning,
      borderColor: darken(0.15, colors.status.warning),
    },
    error: {
      color: colors.white,
      background: colors.status.error,
      borderColor: darken(0.15, colors.status.error),
    },
  },
  Button: {
    default: {
      color: colors.black,
      background: colors.white,
      hoverColor: colors.purple800,
      hoverBackground: darken(0.05, colors.white),
    },
    primary: {
      color: colors.white,
      background: colors.turquoise,
      hoverColor: colors.white,
      hoverBackground: darken(0.05, colors.turquoise),
    },
    danger: {
      color: colors.white,
      background: colors.status.error,
      hoverColor: colors.white,
      hoverBackground: darken(0.05, colors.status.error),
    },
    disabled: {
      color: colors.doveGray,
      background: colors.gallery,
      hoverColor: colors.doveGray,
      hoverBackground: colors.gallery,
    },
  },
};
