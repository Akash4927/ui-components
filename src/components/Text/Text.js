import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { lookupValue } from '../../utils/theme';

const StyledText = styled.span`
  font-family: ${props => props.theme.fontFamily};
  color: ${props => props.theme.textColor};
  font-size: ${props => lookupValue(props, props.theme.fontSizes, props.theme.fontSizes.normal)};
  font-weight: ${props => (props.bold ? '600' : '400')};
  text-transform: ${props => (props.capitalize ? 'uppercase' : 'none')};
  font-style: ${props => (props.italic ? 'italic' : 'normal')}
`;


/**
 * Text! Normal, Large and Extra Large are 0.875em, 2em, and 2.827em respectively.
 * The font is proxima-nova.
 * ```javascript
 * import { Text } from 'weaveworks-ui-components';
 *
 * export default function TextExample() {
 *   return (
 *     <div>
 *       <Text>Normal</Text>
 *
 *       <Text italic>Normal Italic</Text>
 *
 *       <Text bold>Normal Bold</Text>
 *
 *       <Text large>Large</Text>
 *
 *       <Text large italic>Large Italic</Text>
 *
 *       <Text large bold>Large Bold</Text>
 *
 *       <Text xl>Extra Large</Text>
 *
 *       <Text xl italic>Extra Large Italic</Text>
 *
 *       <Text xl bold>Extra Large Bold</Text>
 *     </div>
 *   );
 * }
 * ```
 */
// Working around https://github.com/weaveworks/ui-components/issues/38
const Text = props => (
  <StyledText {...props} />
);

Text.propTypes = {
  /**
   * Italicize the text
   */
  italic: PropTypes.bool,
  /**
   * Make the text bold
   */
  bold: PropTypes.bool,
  /**
   * Default text size; 0.875em
   */
  normal: PropTypes.bool,
  /**
   * Text size will be 2em
   */
  large: PropTypes.bool,
  /**
   * Text size will be 2.827em
   */
  xl: PropTypes.bool,
};

export default Text;
