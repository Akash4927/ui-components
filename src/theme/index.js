import { colors, themeColorsAsScss } from './colors';
import { fontSizes, fontFamilies, themeFontsAsScss } from './fonts';
import { borderRadius, themeBorderRadiiAsScss } from './borders';
import { layers, themeLayersAsScss } from './layers';
import { spacing, themeSpacingsAsScss } from './spacings';
import { overlayIconSize, textColor, themeMiscVarsAsScss } from './misc';
import { boxShadow } from './box-shadows';
import { atoms } from './atoms';

export function themeVarsAsScss() {
  const themeVariables = []
    .concat(themeColorsAsScss())
    .concat(themeLayersAsScss())
    .concat(themeFontsAsScss())
    .concat(themeBorderRadiiAsScss())
    .concat(themeSpacingsAsScss())
    .concat(themeMiscVarsAsScss());
  return `${themeVariables.join('; ')};`;
}

const theme = {
  // Component-specific
  atoms,

  // Box shadows
  boxShadow,

  // Borders
  borderRadius,

  // Colors
  colors,

  // Fonts
  fontFamilies,
  fontSizes,

  // z-index layers
  layers,

  // Spacings
  spacing,

  // Misc
  overlayIconSize,
  textColor,
};

export default theme;
