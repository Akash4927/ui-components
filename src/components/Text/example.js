import React from 'react';

import { Example } from '../../utils/example';

import Text from '.';

export default function TextExample() {
  return (
    <div>
      <Example>
        <Text tiny>Tiny</Text>
        <br />
        <Text tiny italic>
          Tiny Italic
        </Text>
        <br />
        <Text tiny bold>
          Tiny Bold
        </Text>
      </Example>
      <Example>
        <Text small>Small</Text>
        <br />
        <Text small italic>
          Small Italic
        </Text>
        <br />
        <Text small bold>
          Small Bold
        </Text>
      </Example>
      <Example>
        <Text>Normal</Text>
        <br />
        <Text italic>Normal Italic</Text>
        <br />
        <Text bold>Normal Bold</Text>
      </Example>
      <Example>
        <Text large>Large</Text>
        <br />
        <Text large italic>
          Large Italic
        </Text>
        <br />
        <Text large bold>
          Large Bold
        </Text>
      </Example>
      <Example>
        <Text extraLarge>Extra Large</Text>
        <br />
        <Text extraLarge italic>
          Extra Large Italic
        </Text>
        <br />
        <Text extraLarge bold>
          Extra Large Bold
        </Text>
      </Example>
      <Example>
        <Text huge>Huge</Text>
        <br />
        <Text huge italic>
          Huge Italic
        </Text>
        <br />
        <Text huge bold>
          Huge Bold
        </Text>
      </Example>
      <Example>
        <Text>
          Quo non arbitrantur. Et irure comprehenderit iis nescius duis officia
          cernantur. Quae eiusmod voluptatibus, o se magna fugiat quid, culpa
          adipisicing quamquam labore quamquam, dolore ex litteris. Fugiat id
          cernantur, fabulas tamen ullamco cupidatat.Expetendis amet incurreret,
          illum cupidatat occaecat, veniam concursionibus laborum fugiat
          voluptate, te ut familiaritatem. Quo se sint fore quis an eram
          quamquam comprehenderit, an ex sempiternum, ex ubi despicationes.
          Occaecat quae quibusdam cernantur, arbitror tempor appellat ad quo
          noster quo dolor, multos iudicem nostrud. Culpa aliquip nostrud, ubi
          quibusdam iis deserunt, qui e sempiternum iis occaecat quem admodum
          possumus, aut nam exquisitaque, ingeniis ex eram admodum, quibusdam
          reprehenderit quo arbitror, hic multos consequat.
        </Text>
      </Example>
    </div>
  );
}
