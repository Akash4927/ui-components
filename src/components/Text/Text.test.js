import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import theme from '../../../theme';
import Text from './Text';

describe('<Text />', () => {
  it('renders default', () => {
    const tree = renderer.create(<Text theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renderes large', () => {
    const tree = renderer.create(<Text large theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders bold', () => {
    const tree = renderer.create(<Text italic theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders italic', () => {
    const tree = renderer.create(<Text large theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders xl', () => {
    const tree = renderer.create(<Text xl theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
