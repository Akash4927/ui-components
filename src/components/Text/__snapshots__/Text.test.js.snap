// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renderes bold 1`] = `
.c0 {
  font-family: 'proxima-nova',Helvetica,Arial,sans-serif;
  color: hsl(0,0%,10%);
  font-size: 16px;
  font-weight: 600;
  text-transform: none;
  font-style: normal;
}

<span
  className="c0"
/>
`;

exports[`<Text /> renders default 1`] = `
.c0 {
  font-family: 'proxima-nova',Helvetica,Arial,sans-serif;
  color: hsl(0,0%,10%);
  font-size: 16px;
  font-weight: 400;
  text-transform: none;
  font-style: normal;
}

<span
  className="c0"
/>
`;

exports[`<Text /> renders italic 1`] = `
.c0 {
  font-family: 'proxima-nova',Helvetica,Arial,sans-serif;
  color: hsl(0,0%,10%);
  font-size: 16px;
  font-weight: 400;
  text-transform: none;
  font-style: italic;
}

<span
  className="c0"
/>
`;

exports[`<Text /> renders large 1`] = `
.c0 {
  font-family: 'proxima-nova',Helvetica,Arial,sans-serif;
  color: hsl(0,0%,10%);
  font-size: 22px;
  font-weight: 400;
  text-transform: none;
  font-style: normal;
}

<span
  className="c0"
/>
`;
